llrt_modules/src/modules/fs/file_handle.rs

```diff
@@ -0,0 +1,862 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use std::borrow::Cow;
+use std::path::PathBuf;
+
+use either::Either;
+use llrt_utils::encoding::Encoder;
+use llrt_utils::object::ObjectExt;
+use llrt_utils::result::{OptionExt, ResultExt};
+use rquickjs::function::Opt;
+use rquickjs::{Ctx, Error, Exception, FromJs, Null, Object, Result, Value};
+use tokio::io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt, SeekFrom};
+use tokio::{fs::File, task};
+
+use super::{read_file, Stat};
+use crate::buffer::Buffer;
+use crate::utils::array_buffer::ArrayBufferView;
+
+const DEFAULT_BUFFER_SIZE: usize = 16384;
+const DEFAULT_ENCODING: &str = "utf8";
+
+#[rquickjs::class]
+#[derive(rquickjs::class::Trace)]
+pub struct FileHandle {
+    #[qjs(skip_trace)]
+    file: Option<File>,
+    #[qjs(skip_trace)]
+    path: PathBuf,
+}
+
+impl FileHandle {
+    pub fn new(file: File, path: PathBuf) -> Self {
+        Self {
+            file: Some(file),
+            path,
+        }
+    }
+
+    fn file(&self, ctx: &Ctx<'_>) -> Result<&File> {
+        self.file.as_ref().or_throw_msg(ctx, "FileHandle is closed")
+    }
+
+    fn file_mut(&mut self, ctx: &Ctx<'_>) -> Result<&mut File> {
+        self.file.as_mut().or_throw_msg(ctx, "FileHandle is closed")
+    }
+}
+
+#[rquickjs::methods(rename_all = "camelCase")]
+impl FileHandle {
+    async fn chmod(&self, ctx: Ctx<'_>, mode: u32) -> Result<()> {
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            let perm = std::fs::Permissions::from_mode(mode);
+            self.file(&ctx)?
+                .set_permissions(perm)
+                .await
+                .or_throw_msg(&ctx, "Can't modify file permissions")?;
+        }
+        Ok(())
+    }
+
+    async fn chown(&self, ctx: Ctx<'_>, uid: u32, gid: u32) -> Result<()> {
+        #[cfg(unix)]
+        {
+            let path = self.path.clone();
+            task::spawn_blocking(move || std::os::unix::fs::chown(&path, Some(uid), Some(gid)))
+                .await
+                .or_throw(&ctx)?
+                .or_throw_msg(&ctx, "Can't modify file owner")?;
```

Comment:
Do we need to throw twice here?

Reply:
Yeah one is for  the spawn_blocking result and one is for the chown result